
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="canonical" href="https://mcw-mstp.github.io/rna_seq/lect1/lect1/">
      
      <link rel="icon" href="../../../assets/codes_favicon.svg">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-7.3.4">
    
    
      
        <title>Intro to scRNA seq - MCW MSTP Codes</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.db9e7362.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    
      

  


  

  


  <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-4EB03M9P81"),document.addEventListener("DOMContentLoaded",function(){"undefined"!=typeof location$&&location$.subscribe(function(t){gtag("config","G-4EB03M9P81",{page_path:t.pathname})})})</script>
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-4EB03M9P81"></script>


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("../../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#lecture-1" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
        <aside class="md-banner md-announce">
          <div class="md-banner__inner md-announce__inner md-grid md-typeset">
            
    Add your name to the 'People' section!

          </div>
        </aside>
      
    </div>
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="MCW MSTP Codes" class="md-header__button md-logo" aria-label="MCW MSTP Codes" data-md-component="logo">
      
  <img src="../../../assets/cd.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            MCW MSTP Codes
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Intro to scRNA seq
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/mcw-mstp/mcw-mstp.github.io/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        Home
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../single_cell/" class="md-tabs__link md-tabs__link--active">
        Singe Cell Seq
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../abacus/" class="md-tabs__link">
        ABACUS
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../resources/resources/" class="md-tabs__link">
        Resources
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../people/people/" class="md-tabs__link">
        People
      </a>
    </li>
  

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../../contributing/" class="md-tabs__link">
      Contributing
    </a>
  </li>

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="MCW MSTP Codes" class="md-nav__button md-logo" aria-label="MCW MSTP Codes" data-md-component="logo">
      
  <img src="../../../assets/cd.svg" alt="logo">

    </a>
    MCW MSTP Codes
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/mcw-mstp/mcw-mstp.github.io/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_1">
          Home
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Home" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Home
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Welcome
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../news/" class="md-nav__link">
        News
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Singe Cell Seq
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Singe Cell Seq" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Singe Cell Seq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../single_cell/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Intro to scRNA seq
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Intro to scRNA seq
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          ABACUS
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="ABACUS" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          ABACUS
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../abacus/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../abacus_resources/" class="md-nav__link">
        Resources
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Resources
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Resources" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Resources
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../resources/resources/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../resources/pointers/" class="md-nav__link">
        Pointers
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          People
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="People" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          People
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../people/people/" class="md-nav__link">
        About
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_2">
          Individual Pages
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Individual Pages" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          Individual Pages
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../people/smazurchuk/" class="md-nav__link">
        Stephen Mazurchuk
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/" class="md-nav__link">
        Contributing
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/mcw-mstp/mcw-mstp.github.io/edit/main/docs/rna_seq/lect1/lect1.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="lecture-1">Lecture 1</h1>
<p>These slides are part of a workshop by the Cui Lab on 6/21/21 about how to use R to analyze single-cell RNA sequencing data. It starts by going over the basics of scRNA-seq and how it differs from bulk RNA-seq. We'll then download a publicly available dataset and analyze it together using the attached script.</p>
<p>Feel free to look at or <a href="https://mcw0-my.sharepoint.com/:p:/g/personal/smazurchuk_mcw_edu/EdcN_zorz-ZCjm8nmUtcm5wBfR54AZ4CE5--Pu6ePia28w?e=7vKgpg"><strong>download</strong></a> the slides below!</p>
<iframe src="https://mcw0-my.sharepoint.com/personal/smazurchuk_mcw_edu/_layouts/15/Doc.aspx?sourcedoc={3aff0dd7-cf2b-42e6-8e6f-27994b5c9b9c}&amp;action=embedview&amp;wdAr=1.7777777777777777" width="1000px" height="600px" frameborder="0">This is an embedded <a target="_blank" href="https://office.com">Microsoft Office</a> presentation, powered by <a target="_blank" href="https://office.com/webapps">Office</a>.</iframe>
<p><br></p>
<h2 id="results">Results</h2>
<p>Click <a href="../scRNAseq_analysis.html"><strong>here</strong></a> to see the rendered results, or <a href="../scRNAseq_analysis.Rmd"><strong>here</strong></a> to download the <code>.Rmd</code> file used to generate the output.</p>
<h1 id="scrna_seq-analysis">scRNA_seq analysis</h1>
<p>This is an R Markdown document. Code can only be written in chunks,
which are set off by sets of three tick marks (```).</p>
<p>Spaces between chunks can be used to take notes.</p>
<p>To add a new chunk, click the “Insert” icon towards the top right of the
script section of RStudio. Then select “R” from the dropdown menu.
Alternatively, press Command + Option + I (Mac) or Control + Alt + I
(Windows) on your keyboard.</p>
<p>You can name chunks by typing a name after the letter “r” inside the
curly braces. The drop down menu in the bottom left of the script
section of RStudio allows you to skip to any chunk, so naming chunks
makes it easy to keep track of where you want to skip to in your script.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Inside a chunk, anything in a line written after a pound sign/hashtag is a comment and will not be run as code</span>
<span class="c1"># Use comments to take notes in line with your code</span>

<span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;Seurat&quot;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;tidyverse&quot;</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># Downloaded packages are not automatically available for use in R</span>
<span class="c1"># The library() function loads packages for use in one session of R</span>
<span class="c1"># If you restart R, you don&#39;t need to install the packages again, but you do need to library() them</span>

<span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Attaching SeuratObject
</code></pre>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Registered S3 method overwritten by 'cli':
##   method     from         
##   print.boxx spatstat.geom

## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

## ✓ ggplot2 3.3.5     ✓ purrr   0.3.4
## ✓ tibble  3.1.4     ✓ dplyr   1.0.7
## ✓ tidyr   1.1.3     ✓ stringr 1.4.0
## ✓ readr   2.0.1     ✓ forcats 0.5.1

## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># Seurat expects the input files to be called &quot;barcodes.tsv.gz&quot;, &quot;features.tsv.gz&quot;, and &quot;matrix.mtx.gz&quot;</span>
<span class="c1"># We first need to rename the downloaded files to make them match these names exactly</span>

<span class="nf">file.rename</span><span class="p">(</span><span class="s">&quot;matrix_files/GSM3701181_GP33_day30_barcodes.tsv.gz&quot;</span><span class="p">,</span> <span class="s">&quot;matrix_files/barcodes.tsv.gz&quot;</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning in file.rename("matrix_files/GSM3701181_GP33_day30_barcodes.tsv.gz", :
## cannot rename file 'matrix_files/GSM3701181_GP33_day30_barcodes.tsv.gz' to
## 'matrix_files/barcodes.tsv.gz', reason 'No such file or directory'

## [1] FALSE
</code></pre>
<div class="highlight"><pre><span></span><code><span class="nf">file.rename</span><span class="p">(</span><span class="s">&quot;matrix_files/GSM3701181_GP33_day30_genes.tsv.gz&quot;</span><span class="p">,</span> <span class="s">&quot;matrix_files/features.tsv.gz&quot;</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning in file.rename("matrix_files/GSM3701181_GP33_day30_genes.tsv.gz", :
## cannot rename file 'matrix_files/GSM3701181_GP33_day30_genes.tsv.gz' to
## 'matrix_files/features.tsv.gz', reason 'No such file or directory'

## [1] FALSE
</code></pre>
<div class="highlight"><pre><span></span><code><span class="nf">file.rename</span><span class="p">(</span><span class="s">&quot;matrix_files/GSM3701181_GP33_day30_matrix.mtx.gz&quot;</span><span class="p">,</span> <span class="s">&quot;matrix_files/matrix.mtx.gz&quot;</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning in file.rename("matrix_files/GSM3701181_GP33_day30_matrix.mtx.gz", :
## cannot rename file 'matrix_files/GSM3701181_GP33_day30_matrix.mtx.gz' to
## 'matrix_files/matrix.mtx.gz', reason 'No such file or directory'

## [1] FALSE
</code></pre>
<div class="highlight"><pre><span></span><code><span class="n">Cl13.raw</span> <span class="o">&lt;-</span> <span class="nf">Read10X</span><span class="p">(</span><span class="n">data.dir</span> <span class="o">=</span> <span class="s">&quot;matrix_files/&quot;</span><span class="p">)</span>

<span class="c1"># Cl13.raw is currently stored as a matrix object in R (matrix of genes and cells)</span>
<span class="c1"># We can check how many cells and genes are available for analysis with the dim() function</span>

<span class="nf">dim</span><span class="p">(</span><span class="n">Cl13.raw</span><span class="p">)</span>
</code></pre></div>
<pre><code>## [1] 27998  1838
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># 27998  1838</span>

<span class="c1"># dim() always returns the number of rows, then the number of columns in a matrix or data frame</span>
<span class="c1"># scRNA-seq data is always arranged such that rows are genes and columns are cells</span>
<span class="c1"># Therefore, this dataset has 27,998 genes detected across 1,838 cells</span>

<span class="c1"># We need to make a Seurat object from the matrix object Cl13.raw</span>

<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">counts</span> <span class="o">=</span> <span class="n">Cl13.raw</span><span class="p">,</span> <span class="n">project</span> <span class="o">=</span> <span class="s">&quot;Cl13&quot;</span><span class="p">)</span>

<span class="c1"># You can also call dim() on Seurat objects to see how many genes and cells are in your dataset</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>
</code></pre></div>
<pre><code>## [1] 27998  1838
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># 27998  1838</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># Not all cells are usable for analysis, so we need to filter out low quality cells</span>

<span class="c1"># First, have Seurat calculate what percentage of the transcriptome of each cell comes from mitochondrial genes</span>
<span class="c1"># Cells with high percentages of mitochondrial mRNA may be apoptotic</span>
<span class="c1"># Note that all mitochondrial genes being with &quot;mt-&quot;, so Seurat can easily find these genes</span>
<span class="n">Cl13</span><span class="o">$</span><span class="n">percent.mt</span> <span class="o">&lt;-</span> <span class="nf">PercentageFeatureSet</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">Cl13</span><span class="p">,</span> <span class="n">pattern</span> <span class="o">=</span> <span class="s">&quot;^mt-&quot;</span><span class="p">)</span>

<span class="c1"># View the number of unique genes (nFeature_RNA), number of total transcripts (nCount_RNA), and percent of mitochondrial reads (percent.mt) for each cell</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;percent.mt&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Quality%20control-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Notice how some cells have many more unique genes than most cells</span>
<span class="c1"># These cells may be doublets (two cells captured in the same oil droplet and labeled with the same barcode sequence)</span>
<span class="c1"># We need to remove these cells to prevent contamination</span>

<span class="c1"># We generally also remove cells with very few unique genes per cell, as these cells will not provide much information and may decrease the quality of the analysis</span>
<span class="c1"># Cells in this dataset are above the typical minimal threshold (200 genes per cell)</span>

<span class="c1"># Live cells will have a low percentage of mitochondrial mRNA, so we also use this metric to filter cells</span>
<span class="c1"># Typical cutoffs for percent.mt are 5-10% depending on how stringent you want your quality control to be and how many cells you can afford to filter out</span>

<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">subset</span> <span class="o">=</span> <span class="n">nFeature_RNA</span> <span class="o">&gt;</span> <span class="m">200</span> <span class="o">&amp;</span> <span class="n">nFeature_RNA</span> <span class="o">&lt;</span> <span class="m">2500</span> <span class="o">&amp;</span> <span class="n">percent.mt</span> <span class="o">&lt;</span> <span class="m">10</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># Normalize the data - divides the number of reads for each gene in a cell by the total number of reads in that cell</span>
<span class="c1"># Normalization makes reads comparable betwwen cells that had different read depth during sequencing</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">NormalizeData</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>

<span class="c1"># Select highly variable features - these will be used for PCA/UMAP</span>
<span class="c1"># Running PCA/UMAP on all genes is computationally intensive and doesn&#39;t really improve results</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">FindVariableFeatures</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>

<span class="c1"># You can examine the variable features on a plot if you want to see them</span>
<span class="nf">LabelPoints</span><span class="p">(</span><span class="nf">VariableFeaturePlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">),</span>
            <span class="n">points</span> <span class="o">=</span> <span class="nf">head</span><span class="p">(</span><span class="nf">VariableFeatures</span><span class="p">(</span><span class="n">Cl13</span><span class="p">),</span> <span class="m">10</span><span class="p">))</span>
</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis

## Warning: Removed 14339 rows containing missing values (geom_point).
</code></pre>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Normalization-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Load genes for cell cycle regression</span>
<span class="n">CC_genes</span> <span class="o">&lt;-</span> <span class="n">Seurat</span><span class="o">::</span><span class="n">cc.genes.updated.2019</span>

<span class="c1"># These genes are currently in human gene format (all caps)</span>
<span class="c1"># We need to change them to mouse gene format (first letter caps, all other letters lowercase)</span>
<span class="n">CC_genes</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">CC_genes</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="nf">paste0</span><span class="p">(</span>
  <span class="nf">substring</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">1</span><span class="p">),</span>
  <span class="nf">tolower</span><span class="p">(</span><span class="nf">substring</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="m">2</span><span class="p">)))})</span>

<span class="c1"># Calculate the cell cycle phase of cells</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">CellCycleScoring</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">s.features</span> <span class="o">=</span> <span class="n">CC_genes</span><span class="o">$</span><span class="n">s.genes</span><span class="p">,</span> <span class="n">g2m.features</span> <span class="o">=</span> <span class="n">CC_genes</span><span class="o">$</span><span class="n">g2m.genes</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning: The following features are not present in the object: Pimreg, Jpt1, not
## searching for symbol synonyms
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># Note that not all cell cycle genes will always be found in each dataset</span>

<span class="c1"># Scale the data - scales the expression of each gene across all cells to a range centered at 0 (usually -2 to +2)</span>
<span class="c1"># While scaling, we&#39;ll also regress out cell cycle phase scores so these don&#39;t impact data scaling</span>
<span class="c1"># By default, Seurat only scales variable genes, which we found with the FindVariableFeatures() function earlier</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">ScaleData</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">vars.to.regress</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;S.Score&quot;</span><span class="p">,</span> <span class="s">&quot;G2M.Score&quot;</span><span class="p">))</span>
</code></pre></div>
<pre><code>## Regressing out S.Score, G2M.Score

## Centering and scaling data matrix
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># If you want to scale all genes, for example to generate large heatmaps later, you need to specify this</span>
<span class="c1"># We aren&#39;t doing this today for the sake of time, but the command to do so is below</span>
<span class="c1"># Cl13 &lt;- ScaleData(Cl13, features = rownames(Cl13), vars.to.regress = c(&quot;S.Score&quot;, &quot;G2M.Score&quot;))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># We need to generate a PCA plot before we contruct a UMAP plot</span>
<span class="c1"># By default, 50 PCs are calculated; you can choose to set fewer PCs manually</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">npcs</span> <span class="o">=</span> <span class="m">30</span><span class="p">)</span>
</code></pre></div>
<pre><code>## PC_ 1 
## Positive:  Ccl5, Gzmb, Gzma, Cx3cr1, Zeb2, Ahnak, Lgals1, Ccl4, Klre1, S1pr5 
##     Klrc1, Crip1, Rap1b, Id2, Emp3, Lgals3, Cd48, S100a6, Klrd1, Ifngr1 
##     S100a4, Spn, Txk, Il18rap, Klrc2, Vim, Calm1, Actb, Klrb1c, Klrg1 
## Negative:  Rps28, Rps18, Rpl13, Slamf6, Rps19, Tcf7, Rplp1, Rpl39, Rps29, Rps20 
##     Rps9, Rplp2, Rps15a, Xcl1, Rpl32, Rpl37, Rpl23, Rps16, Rps7, Id3 
##     Il7r, Rps6, Rpl36a, Rpl35, Rpl36, Ltb, Rplp0, Rpl12, Rpl41, Rps5 
## PC_ 2 
## Positive:  Lgals1, Ly6c2, Cx3cr1, Zeb2, Tmsb10, Ahnak, Atp1b3, Klrd1, Klf2, Il18rap 
##     S100a10, Klrk1, Klre1, S1pr5, Klrc1, Il18r1, Klf3, Vim, Rap1b, Emp3 
##     Itgb1, H2afz, Spn, Tagln2, Rasgrp2, S1pr1, S100a4, Klrg1, Pycard, Rpl41 
## Negative:  Cd7, Rgs1, Nr4a2, Lag3, 2900026A02Rik, Cd160, Gzmk, Tox, Cd244, Sept4 
##     Sh2d2a, AW112010, Ptger4, Cxcr6, Pdcd1, Ccl4, Ccl3, Lax1, Abi3, Ikzf2 
##     Serpina3g, Abcb9, Maf, Adgrg1, Adam19, Chn2, Cd200r1, Actn2, Cd27, Tnfrsf9 
## PC_ 3 
## Positive:  Zeb2, Klre1, S1pr5, Klra9, Rps29, PISD, Gzma, Klrc2, Klrb1c, Kcnj8 
##     Tcf7, Klf2, Txk, Il7r, Id3, Rps28, Klrk1, Kcnq1ot1, Il18rap, Klrd1 
##     Bcl2l11, Itgb1, Tnfsf8, Arl4c, Xcl1, Zfp36l1, Bcl2, Klra3, Rpl36, As3mt 
## Negative:  Actg1, Sub1, Pfn1, S100a6, S100a11, S100a4, Crip1, Coro1a, Lgals3, Lsp1 
##     Glipr2, Gapdh, Ccnb2, Rbm3, Txn1, Arpc4, Actr3, Cotl1, Vim, Ifi27l2a 
##     Prdx1, Lrrc58, Calm1, AW112010, Anxa2, Ube2c, Ywhah, S100a10, Arpc5, Rom1 
## PC_ 4 
## Positive:  Ifit3, Ifit1, Rsad2, Isg15, Usp18, Ifit3b, Irf7, Ifi204, Slfn8, Slfn5 
##     Rtp4, Pydc3, Pydc4, Gm4955, Ifih1, Oas3, Isg20, Trim30a, Iigp1, Cxcl10 
##     Stat1, Herc6, Cmpk2, Zbp1, Pyhin1, Oasl2, Mx1, Mnda, Bst2, Oasl1 
## Negative:  Pfn1, Actg1, Rps7, Gm9493, Tnfrsf9, S100a6, Gapdh, Rplp0, Rpl10, Coro1a 
##     Rpl3, Rps6, Adam19, Rps15a, Rpl4, Rps2, S100a4, Rpl41, Snhg6, Rpl36a 
##     Rpl39, Pglyrp1, Chchd10, Rpl32, S100a11, Rpl22l1, Etfb, Ggt1, Sub1, Ctla2a 
## PC_ 5 
## Positive:  Hmmr, Birc5, Ccna2, Kif4, Ckap2l, Casc5, Cenpf, Esco2, Neil3, 2810417H13Rik 
##     Rrm2, Cdca8, Hist1h1b, Aspm, Kif14, Kif22, Kifc1, Nusap1, Tpx2, Cdca3 
##     Nsl1, Mxd3, Cenph, Pbk, Hist1h2ap, Cdc20, Kif2c, Ccnb2, Bub1, Rad54b 
## Negative:  Hmgb2, Smc4, Ccl5, Rpl32, Rpl39, Rpl37, Rpl13, Rpl10a, Rps7, Rps15a 
##     Rpl23a, Samhd1, Rpl18, Rpl3, Ifit1bl1, Rps6, Rpl36, Rpl41, Rps18, Plac8 
##     Rplp2, Rps16, Rps10, Rplp1, Epsti1, Rplp0, Ifit1, Crip1, Rps19, Rpl4
</code></pre>
<div class="highlight"><pre><span></span><code><span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">FindNeighbors</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Computing nearest neighbor graph

## Computing SNN
</code></pre>
<div class="highlight"><pre><span></span><code><span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">FindClusters</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">resolution</span> <span class="o">=</span> <span class="m">0.3</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 1790
## Number of edges: 101145
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8835
## Number of communities: 5
## Elapsed time: 0 seconds
</code></pre>
<div class="highlight"><pre><span></span><code><span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">RunUMAP</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
## To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
## This message will be shown once per session

## 10:07:12 UMAP embedding parameters a = 0.9922 b = 1.112

## 10:07:12 Read 1790 rows and found 30 numeric columns

## 10:07:12 Using Annoy for neighbor search, n_neighbors = 30

## 10:07:12 Building Annoy index with metric = cosine, n_trees = 50

## 0%   10   20   30   40   50   60   70   80   90   100%

## [----|----|----|----|----|----|----|----|----|----|

## **************************************************|
## 10:07:13 Writing NN index file to temp file /tmp/RtmpMkSBB6/file311af47e6ae98
## 10:07:13 Searching Annoy index using 1 thread, search_k = 3000
## 10:07:13 Annoy recall = 100%
## 10:07:13 Commencing smooth kNN distance calibration using 1 thread
## 10:07:14 Initializing from normalized Laplacian + noise
## 10:07:14 Commencing optimization for 500 epochs, with 74660 positive edges
## 10:07:16 Optimization finished
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># Show the UMAP plot</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/UMAP-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Clusters are numbered by decreasing size</span>

<span class="c1"># There isn&#39;t a set number of PCs to use for PCA or UMAP construction</span>
<span class="c1"># In general, PCs are set between 10 and 50</span>
<span class="c1"># It&#39;s usually better to use more PCs than fewer, as too few PCs will drastically alter the data whereas extra PCs will hardly change the overall structure of the data</span>
<span class="c1"># The larger the dataset, the more PCs should be used</span>

<span class="c1"># Resolution in FindClusters() doesn&#39;t affect the shape of the clusters but does affect the coloring (cluster assignment) of cells</span>
<span class="c1"># Look at what happens when we change only the resolution and recreate the UMAP plot</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">FindClusters</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">resolution</span> <span class="o">=</span> <span class="m">0.2</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 1790
## Number of edges: 101145
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9131
## Number of communities: 3
## Elapsed time: 0 seconds
</code></pre>
<div class="highlight"><pre><span></span><code><span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/UMAP-2.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Alternatively, we can color the cells based on their cell cycle phase instead of their cluster identity</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;Phase&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/UMAP-3.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Always test to make sure that your cell populations are pure, even if you sequenced FACS-sorted cells (sorting is never perfect)</span>
<span class="c1"># In this case, the cells are supposed to be LCMV GP33-specific CD8 T cells</span>
<span class="c1"># We can examine identity genes with different plots</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cd8a&quot;</span><span class="p">,</span> <span class="s">&quot;Cd3e&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Cell%20purity-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">FeaturePlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cd8a&quot;</span><span class="p">,</span> <span class="s">&quot;Cd3e&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Cell%20purity-2.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># All clusters express high levels of Cd3e and CD8a, suggesting that these cells are CD8 T cells</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># If we don&#39;t know what genes are important, we can analyze differentially expressed genes (DEGs) in each cluster</span>
<span class="n">Cl13_markers</span> <span class="o">&lt;-</span> <span class="nf">FindAllMarkers</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">only.pos</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Calculating cluster 0

## For a more efficient implementation of the Wilcoxon Rank Sum Test,
## (default method for FindMarkers) please install the limma package
## --------------------------------------------
## install.packages('BiocManager')
## BiocManager::install('limma')
## --------------------------------------------
## After installation of limma, Seurat will automatically use the more 
## efficient implementation (no further action necessary).
## This message will be shown once per session

## Calculating cluster 1

## Calculating cluster 2
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># Setting only.pos = T tells Seurat to only look for genes that are upregulated in each cluster</span>

<span class="c1"># Let&#39;s examine the top 10 DEGs per cluster</span>
<span class="n">Cl13_markers</span> <span class="o">%&gt;%</span> <span class="nf">group_by</span><span class="p">(</span><span class="n">cluster</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">top_n</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="n">wt</span> <span class="o">=</span> <span class="n">avg_log2FC</span><span class="p">)</span>
</code></pre></div>
<pre><code>## # A tibble: 30 × 7
## # Groups:   cluster [3]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene 
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;
##  1 1.44e-165       2.97 0.712 0.116 4.03e-161 0       Cd7  
##  2 4.87e- 88       1.47 0.656 0.211 1.36e- 83 0       Cxcr6
##  3 1.62e- 83       1.26 0.706 0.255 4.54e- 79 0       Gzmk 
##  4 1.60e- 66       1.74 0.699 0.37  4.49e- 62 0       Rgs1 
##  5 1.07e- 64       1.22 0.703 0.346 2.98e- 60 0       Lag3 
##  6 1.79e- 63       1.65 0.867 0.564 5.01e- 59 0       Ccl4 
##  7 5.34e- 61       1.81 0.695 0.351 1.49e- 56 0       Ccl3 
##  8 8.52e- 53       1.29 0.433 0.127 2.39e- 48 0       Maf  
##  9 1.35e- 51       1.24 0.708 0.408 3.78e- 47 0       Nr4a2
## 10 8.70e- 49       1.22 0.418 0.13  2.43e- 44 0       Cd160
## # … with 20 more rows
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># Note that gene names don&#39;t always match protein names (Pdcd1 = PD-1, Slamf6 = Ly108)</span>

<span class="c1"># We can plot some of these genes to visualize how specific they are to each cluster</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cxcr6&quot;</span><span class="p">,</span> <span class="s">&quot;Nr4a2&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;Zeb2&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">,</span> <span class="s">&quot;Il7r&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Differential%20gene%20expression%20analysis-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">DotPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cxcr6&quot;</span><span class="p">,</span> <span class="s">&quot;Nr4a2&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;Zeb2&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">,</span> <span class="s">&quot;Il7r&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Differential%20gene%20expression%20analysis-2.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">FeaturePlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cxcr6&quot;</span><span class="p">,</span> <span class="s">&quot;Nr4a2&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;Zeb2&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">,</span> <span class="s">&quot;Il7r&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Differential%20gene%20expression%20analysis-3.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">RidgePlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cxcr6&quot;</span><span class="p">,</span> <span class="s">&quot;Nr4a2&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;Zeb2&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">,</span> <span class="s">&quot;Il7r&quot;</span><span class="p">))</span>
</code></pre></div>
<pre><code>## Picking joint bandwidth of 0.144

## Picking joint bandwidth of 0.226

## Picking joint bandwidth of 0.0622

## Picking joint bandwidth of 0.0646

## Picking joint bandwidth of 0.0817

## Picking joint bandwidth of 0.086
</code></pre>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Differential%20gene%20expression%20analysis-4.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Using our knowledge of CD8 T cell biology from the literature and our own experiments, we can determine cluster identities:</span>
<span class="c1"># Cluster 0 - Exhausted cells</span>
<span class="c1"># Cluster 1 - Effector cells</span>
<span class="c1"># Cluster 2 - Progenitor cells</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># To make figures easier to follow, we can rename the clusters in our Seurat object</span>
<span class="n">Cl13</span> <span class="o">&lt;-</span> <span class="nf">RenameIdents</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span>
                     <span class="s">&quot;0&quot;</span> <span class="o">=</span> <span class="s">&quot;Exhausted&quot;</span><span class="p">,</span>
                     <span class="s">&quot;1&quot;</span> <span class="o">=</span> <span class="s">&quot;Effector&quot;</span><span class="p">,</span>
                     <span class="s">&quot;2&quot;</span> <span class="o">=</span> <span class="s">&quot;Progenitor&quot;</span><span class="p">)</span>

<span class="c1"># Check to make sure the new names are okay</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">)</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Rename%20clusters-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Make a folder for figures</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;Figures&quot;</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Warning in dir.create("Figures"): 'Figures' already exists
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># The ggsave() function saves plots into a directory you want in a file format you specify</span>
<span class="c1"># ggsave() automatically saves the last plot run in R, so place this command directly after the plot you want to save and run them together</span>
<span class="c1"># Remember to include the &quot;Figures&quot; in the file path inside ggsave()</span>
<span class="c1"># Tip: In RStudio, you can press the Tab key inside quotation marks to automatically list and select file paths</span>

<span class="c1"># UMAP</span>
<span class="c1"># Most plotting functions in Seurat allow you to easily change the colors of the clusters as well</span>
<span class="c1"># Remember that colors are specified in the order that the clusters are listed (by default, clusters are arranged in alphabetical order)</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">cols</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;gray50&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;skyblue&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Save%20figures-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figures/UMAP.png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Saving 7 x 5 in image
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># dpi specifies the image resolution for PNG files; always save at 300 dpi (the default is 72 dpi, which is lower resolution and not acceptable for publications)</span>

<span class="c1"># UMAP colored by cell cycle phase</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;Phase&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Save%20figures-2.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figures/UMAP_cell_cycle.png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Saving 7 x 5 in image
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># FeaturePlot to verify cell purity</span>
<span class="nf">FeaturePlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Cd3e&quot;</span><span class="p">,</span> <span class="s">&quot;Cd8a&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Save%20figures-3.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figures/Cell_purity_FeaturePlot.png&quot;</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Saving 7 x 5 in image
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># VlnPlot of major markers defining each subset</span>
<span class="c1"># Rememer to keep the colors you pick consistent if you don&#39;t want to use the default colors</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Pdcd1&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">),</span> <span class="n">cols</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;gray50&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;skyblue&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Save%20figures-4.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figures/Cluster_markers_VlnPlot.png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">)</span>
</code></pre></div>
<pre><code>## Saving 7 x 5 in image
</code></pre>
<div class="highlight"><pre><span></span><code><span class="c1"># DotPlot of differentially expressed genes from FindAllMarkers() that may be of interest</span>
<span class="c1"># We&#39;ll manually set the height and width of this file so the text on the x-axis fits properly</span>
<span class="nf">DotPlot</span><span class="p">(</span><span class="n">Cl13</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Pdcd1&quot;</span><span class="p">,</span> <span class="s">&quot;Cxcr6&quot;</span><span class="p">,</span> <span class="s">&quot;Nr4a2&quot;</span><span class="p">,</span> <span class="s">&quot;Cx3cr1&quot;</span><span class="p">,</span> <span class="s">&quot;S1pr5&quot;</span><span class="p">,</span> <span class="s">&quot;Tbx21&quot;</span><span class="p">,</span> <span class="s">&quot;Slamf6&quot;</span><span class="p">,</span> <span class="s">&quot;Il7r&quot;</span><span class="p">,</span> <span class="s">&quot;Tcf7&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="" src="../scRNAseq_analysis_files/figure-gfm/Save%20figures-5.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figures/Cluster_markers_DotPlot.png&quot;</span><span class="p">,</span> <span class="n">dpi</span> <span class="o">=</span> <span class="m">300</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">9</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="m">5</span><span class="p">)</span>

<span class="c1"># 5 figures in 1 hour</span>
<span class="c1"># Price: $0</span>

<span class="c1"># Further analyses can be conducted with the following packages:</span>
<span class="c1">#   Monocle - Psuedotime trajectory analysis</span>
<span class="c1">#   VeloCyto - RNA Velocity</span>
<span class="c1">#   SCENIC - regulon (transcription factor gene regulatory network) analysis</span>
<span class="c1">#   CellPhoneDB - Cell-cell receptor-ligand interactions (currently only available as a Python package)</span>
<span class="c1">#   ggpubr - easy statistical analysis for violin or box plots</span>
<span class="c1">#   novoSpaRc - de novo or atlas-based spatial reconstruction of scRNA-seq data (currently only available as a Python package)</span>
</code></pre></div>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../../single_cell/" class="md-footer__link md-footer__link--prev" aria-label="Previous: About" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              About
            </div>
          </div>
        </a>
      
      
        
        <a href="../../../abacus/" class="md-footer__link md-footer__link--next" aria-label="Next: About" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              About
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        
        
      </div>
      
  <div class="md-footer-social">
    
      
      
        
        
      
      <a href="https://github.com/mcw-mstp" target="_blank" rel="noopener" title="github.com" class="md-footer-social__link">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
      </a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs"], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../../assets/javascripts/workers/search.8397ff9e.min.js", "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.1e84347e.min.js"></script>
      
    
  </body>
</html>